solr/core/src/java/org/apache/solr/util/EnvUtils.java

```diff
@@ -0,0 +1,279 @@
+  private static final Map<String, String> CUSTOM_MAPPINGS = new HashMap<>();
+  private static boolean initialized = false;
+
+  static {
+    try {
+      Properties props = new Properties();
+      try (InputStream stream =
+          EnvUtils.class.getClassLoader().getResourceAsStream("EnvToSyspropMappings.properties")) {
+        props.load(new InputStreamReader(Objects.requireNonNull(stream), StandardCharsets.UTF_8));
+        for (String key : props.stringPropertyNames()) {
+          CUSTOM_MAPPINGS.put(key, props.getProperty(key));
+        }
+        init(false);
+      }
+    } catch (IOException e) {
+      throw new SolrException(
+          SolrException.ErrorCode.INVALID_STATE, "Failed loading env.var->properties mapping", e);
+    }
+  }
+
+  /**
+   * Get Solr's mutable copy of all environment variables.
+   *
+   * @return sorted map of environment variables
+   */
+  public static SortedMap<String, String> getEnvs() {
+    return ENV;
+  }
+
+  /** Get a single environment variable as string */
+  public static String getEnv(String key) {
+    return ENV.get(key);
+  }
+
+  /** Get a single environment variable as string, or default */
+  public static String getEnv(String key, String defaultValue) {
+    return ENV.getOrDefault(key, defaultValue);
+  }
+
+  /** Get an environment variable as long */
+  public static long getEnvAsLong(String key) {
+    return Long.parseLong(ENV.get(key));
+  }
+
+  /** Get an environment variable as long, or default value */
+  public static long getEnvAsLong(String key, long defaultValue) {
+    String value = ENV.get(key);
+    if (value == null) {
+      return defaultValue;
+    }
+    return Long.parseLong(value);
+  }
+
+  /** Get an env var as boolean */
+  public static boolean getEnvAsBool(String key) {
+    return StrUtils.parseBool(ENV.get(key));
+  }
+
+  /** Get an env var as boolean, or default value */
+  public static boolean getEnvAsBool(String key, boolean defaultValue) {
+    String value = ENV.get(key);
+    if (value == null) {
+      return defaultValue;
+    }
+    return StrUtils.parseBool(value);
+  }
+
+  /** Get comma separated strings from env as List */
+  public static List<String> getEnvAsList(String key) {
+    return getEnv(key) != null ? stringValueToList(getEnv(key)) : null;
+  }
+
+  /** Get comma separated strings from env as List */
+  public static List<String> getEnvAsList(String key, List<String> defaultValue) {
+    return ENV.get(key) != null ? getEnvAsList(key) : defaultValue;
+  }
+
+  /** Set an environment variable */
+  public static void setEnv(String key, String value) {
+    ENV.put(key, value);
+  }
+
+  /** Set all environment variables */
+  public static synchronized void setEnvs(Map<String, String> env) {
+    ENV.clear();
+    ENV.putAll(env);
+  }
+
+  /** Get all Solr system properties as a sorted map */
+  public static SortedMap<String, String> getProps() {
```

Comment:
I assume sorting is for making the results pretty if printed.  But can we assume the caller is going to print them?  Not sure why we need this a tall actually.

Reply:
I don't have a strong defence, but it is convenient to work with both in actual CLI/UI, in debug print and in tests, especially since the list of vars is so long.